src/audio/igo_nr/igo_nr.c

```diff
@@ -138,43 +186,57 @@ static void igo_nr_capture_s24(struct comp_data *cd,
 static void igo_nr_capture_s32(struct comp_data *cd,
 			       const struct audio_stream *source,
 			       struct audio_stream *sink,
-			       int32_t src_frames,
-			       int32_t sink_frames)
+			       int32_t frames)
 {
 	int32_t in_nch = source->channels;
 	int32_t out_nch = sink->channels;
+	int32_t min_nch = MIN(in_nch, out_nch);
```

Comment:
are we expecting mixed channels? I feel like this is a case we just avoid, by rejecting mismatch in params

Reply:
rejection would be too harsh. We could still deal with the common channels.

Reply:
but then we are leaving the channels to re-read old data

Reply:
It's only copying the data whose index < `min_nch`. The channels whose index >= `min_nch` are don't-care. 

For `in_nch > out_nch`, the remaining source channels are ignored.
For `in_nch < out_nch`, the remaining sink channels will be muted.

Reply:
for case 2 i dont see you writing 0s two any channels > min_ch. In fact i see you advancing the sink the same as the source with regards to index which likely means you would have data leak across channels in a case where out_ch > in_ch. Please simply put in the restriction to enforce `in_ch == out_ch` on params step so we can remove this logic? What does this gain?

Reply:
I agree. A simple restriction in `.params` suffices to remove all the channel mismatch checking. Will add in the next force push.

Reply:
Fix has been applied in 29ea7c65a75ef13cd92f76fcaa7e78bb4fb1dc41. Please review again @cujomalainey .